### DISCUSSION TITLE
My Flask project isn't working now

### DISCUSSION THREAD

**Discussion by cleverpanther** [community/community#134428] [Programming Help] [answered]
### Body

Hello, everyone.
Flask is the first for me so I wrote a simple code using this framework. But there is an error now. The error message is following:
OSError: [WinError 10038] An operation was attempted on something that is not a socket
How can I solve this problem?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by aimaster-dev** [comment, marked as answer]
In the past, I faced this project and it took 1 day to fix this problem. It's because the corruption of winsock2. To solve this problem you need to reset the net settings using the following command:

netsh winsock reset catalog
netsh int ipv4 reset reset.log
netsh int ipv6 reset reset.log

**Reply by cleverpanther** [reply]
Great! It's working well. Thanks

**Reply by aijiultimate-2** [reply]
where do i run this command

**Reply by aijiultimate-2** [reply]
it is still not working

**Comment by cleverpanther** [comment]
But I wanna know why this happens clearly. Can you explain about this?

**Reply by aijiultimate-2** [reply]
this is the flask each time i run it i recive url not found import os
from flask import Flask, jsonify, request, send_from_directory, render_template
from werkzeug.utils import secure_filename

# --- Config ---
UPLOAD_FOLDER = "videos"
ALLOWED_EXTENSIONS = {"mp4", "webm", "ogg"}  # Allowed video types
os.makedirs(UPLOAD_FOLDER, exist_ok=True)

MY_API_2 = os.environ.get('MY_API_2')
print("MY_API_2:", MY_API_2)  # Debugging

app = Flask(__name__, static_folder="static", static_url_path="/static", template_folder="templates")
app.config["UPLOAD_FOLDER"] = UPLOAD_FOLDER

# --- In-memory storage ---
items = []
users = {}
videos = []  # Store video metadata

# --- Helper function ---
def allowed_file(filename):
    return "." in filename and filename.rsplit(".", 1)[1].lower() in ALLOWED_EXTENSIONS

# --- User Registration ---
@app.route('/api/register', methods=['POST'])
def register():
    data = request.get_json()
    username = data.get('username')
    email = data.get('email')
    password = data.get('password')
    if not username or not email or not password:
        return jsonify({"msg": "All fields required."}), 400
    if email in users:
        return jsonify({"msg": "Email already registered."}), 400
    users[email] = {"username": username, "password": password}
    return jsonify({"msg": "Account created successfully."}), 200

# --- User Login ---
@app.route('/api/login', methods=['POST'])
def login():
    data = request.get_json()
    email = data.get('email')
    password = data.get('password')
    user = users.get(email)
    if not user or user['password'] != password:
        return jsonify({"msg": "Invalid email or password."}), 401
    return jsonify({"msg": "Login successful.", "username": user['username']}), 200

# --- Search Videos ---
@app.route('/api/search', methods=['GET'])
def search():
    q = request.args.get('q', '').lower()
    results = [v for v in videos if q in v['title'].lower() or q in v['description'].lower()]
    return jsonify(results), 200

# --- CRUD Items ---
@app.route('/items', methods=['GET'])
def get_items():
    return jsonify(items), 200

@app.route('/items', methods=['POST'])
def add_item():
    new_item = request.json
    items.append(new_item)
    return jsonify(new_item), 201

@app.route('/items/<int:index>', methods=['GET'])
def get_item(index):
    try:
        return jsonify(items[index]), 200
    except IndexError:
        return jsonify({'error': 'Item not found'}), 404

@app.route('/items/<int:index>', methods=['PUT'])
def update_item(index):
    try:
        items[index].update(request.json)
        return jsonify(items[index]), 200
    except IndexError:
        return jsonify({'error': 'Item not found'}), 404

@app.route('/items/<int:index>', methods=['DELETE'])
def delete_item(index):
    try:
        deleted_item = items.pop(index)
        return jsonify(deleted_item), 204
    except IndexError:
        return jsonify({'error': 'Item not found'}), 404

# --- Upload Video ---
@app.route('/api/upload', methods=['POST'])
def upload_video():
    if "file" not in request.files:
        return jsonify({"msg": "No file part"}), 400

    file = request.files["file"]
    title = request.form.get("title", "Untitled Video")
    description = request.form.get("description", "")

    if file.filename == "":
        return jsonify({"msg": "No selected file"}), 400

    if file and allowed_file(file.filename):
        filename = secure_filename(file.filename)
        save_path = os.path.join(app.config["UPLOAD_FOLDER"], filename)
        file.save(save_path)

        video_url = f"/videos/{filename}"
        videos.append({"title": title, "description": description, "url": video_url})

        return jsonify({"msg": "Video uploaded successfully.", "url": video_url}), 201
    else:
        return jsonify({"msg": "Invalid file type."}), 400

# --- Serve Videos ---
@app.route('/videos/<path:filename>')
def serve_video(filename):
    return send_from_directory(app.config["UPLOAD_FOLDER"], filename)

# --- Serve Home Page ---
@app.route('/')
def serve_home():
    return render_template('website.html')

if __name__ == '__main__':
    port = int(os.environ.get('PORT', 5000))
    app.run(host='0.0.0.0', port=port)

**Reply by aijiultimate-2** [reply]
please i need a fix
